modules/gapi/src/backends/fluid/gfluidcore_func.simd.hpp

```diff
@@ -1401,70 +1444,70 @@ CV_ALWAYS_INLINE int mulc_scale_simd_c3(const SRC in[],
-CV_ALWAYS_INLINE void mulc_scale_simd_common_impl(const SRC* inx,
-                                                  uchar* outx, const v_float32& sc,
-                                                  const v_float32& scale, const int nlanes)
+template<typename oper_tag, typename SRC>
+CV_ALWAYS_INLINE void arithmOpScalarScaled_simd_common_impl(oper_tag op, const SRC* inx,
+                                                            uchar* outx, const v_float32& v_scalar,
+                                                            const v_float32& v_scale, const int nlanes)
 {
     v_float32 a1 = vg_load_f32(inx);
     v_float32 a2 = vg_load_f32(&inx[nlanes/4]);
     v_float32 a3 = vg_load_f32(&inx[nlanes/2]);
     v_float32 a4 = vg_load_f32(&inx[3 * nlanes/4]);
 
-    vx_store(outx, v_pack_u(v_pack(v_round(scale * a1* sc),
-                                   v_round(scale * a2* sc)),
-                            v_pack(v_round(scale * a3* sc),
-                                   v_round(scale * a4* sc))));
+    vx_store(outx, v_pack_u(v_pack(v_round(oper_scaled(op, a1, v_scalar, v_scale)),
+                                   v_round(oper_scaled(op, a2, v_scalar, v_scale))),
+                            v_pack(v_round(oper_scaled(op, a3, v_scalar, v_scale)),
+                                   v_round(oper_scaled(op, a4, v_scalar, v_scale)))));
 }
 
 //-------------------------------------------------------------------------------------------------
 
-template<typename SRC>
-CV_ALWAYS_INLINE void mulc_scale_simd_common_impl(const SRC* inx,
-                                                  float* outx, const v_float32& sc,
-                                                  const v_float32& scale, const int)
+template<typename oper_tag, typename SRC>
+CV_ALWAYS_INLINE void arithmOpScalarScaled_simd_common_impl(oper_tag op, const SRC* inx,
+                                                            float* outx, const v_float32& v_scalar,
+                                                            const v_float32& v_scale, const int)
 {
-    v_float32 a1 = vg_load_f32(inx);
-    vx_store(outx, scale * a1* sc);
+    v_float32 a = vg_load_f32(inx);
+    vx_store(outx, oper_scaled(op, a, v_scalar, v_scale));
 }
 
 //-------------------------------------------------------------------------------------------------
 
-template<typename SRC, typename DST>
-CV_ALWAYS_INLINE int mulc_scale_simd_common(const SRC in[],
-                                            const float scalar[], DST out[],
-                                            const int length, const float _scale)
+template<typename oper_tag, typename SRC, typename DST>
+CV_ALWAYS_INLINE int arithmOpScalarScaled_simd_common(oper_tag op, const SRC in[],
+                                                      const float scalar[], DST out[],
+                                                      const int length, const float scale)
 {
     constexpr int nlanes = vector_type_of_t<DST>::nlanes;
```

Comment:
It's not nMaxLane. It is number of numeric (8-bit, 16-bit or 32-bit) that fit to SIMD vector. Since we use dynamic dispatching here and function is templated we cannot know for sure at any particular moment with SIMD vector of what bitness we are dealing with. Also we cannot know for sure what bitness of numbers we will load to this SIMD vector.
`nlanes` changes dynamically depending on data types and vector types. So it must be calculated dynamically.